postgresql-17.yaml

```diff
@@ -407,6 +417,96 @@ subpackages:
+            --with-openssl \
+            --with-libedit-preferred \
+            --with-uuid=e2fs \
+            --with-libxml \
+            --with-libxslt \
+            --with-icu \
+            --with-lz4 \
+            --with-zstd \
+            --with-gssapi \
+            --with-ldap \
+            --with-llvm \
+            --with-python \
+            --enable-thread-safety
+      - uses: autoconf/make
+      - uses: autoconf/make-install
+      - runs: |
+          savedir="${{targets.contextdir}}.sav"
+          mkdir -p $savedir/usr/lib/postgresql${{vars.major-version}}
+          mkdir -p $savedir/usr/share/postgresql${{vars.major-version}}/extension
+          mv ${{targets.contextdir}}/usr/share/postgresql${{vars.major-version}}/extension/plpython* $savedir/usr/share/postgresql${{vars.major-version}}/extension/
+          mv ${{targets.contextdir}}/usr/lib/postgresql${{vars.major-version}}/plpython* $savedir/usr/lib/postgresql${{vars.major-version}}/
+          rm -rf "${{targets.contextdir}}"
+          mv "$savedir" "${{targets.contextdir}}"
+      - uses: strip
+    test:
+      environment:
+        contents:
+          packages:
+            - postgresql-${{vars.major-version}}-client
+            - postgresql-${{vars.major-version}}
+            - shadow
+            - sudo-rs
+            - wait-for-it
+            - python-${{range.key}}
+        environment:
+          PGDATA: /tmp/plpython_test_db
+          PGUSER: testuser
+      pipeline:
+        - uses: test/tw/ldd-check
+        - name: "Test plpython database functionality"
+          runs: |
+            echo "=== Setting up test database for plpython ==="
+            useradd $PGUSER
+            sudo -u $PGUSER initdb -D $PGDATA 2>&1
+            sudo -u $PGUSER pg_ctl -D $PGDATA -l /tmp/plpython_logfile start 2>&1
+            wait-for-it localhost:5432 -t 5 2>&1
+            createdb plpython_testdb 2>&1
+
+            psql -d plpython_testdb -c "CREATE EXTENSION plpython3u;"
+            psql -d plpython_testdb -c "
+            CREATE OR REPLACE FUNCTION test_plpython_basic() RETURNS text AS \$\$
+            return 'Hello from plpython3!'
+            \$\$ LANGUAGE plpython3u;" 2>&1
+            PLPYTHON_STATEMENT=$(psql -t -A -d plpython_testdb -c "SELECT test_plpython_basic();")
+            [ "$PLPYTHON_STATEMENT" = "Hello from plpython3!" ]
+
+            # Create a function that returns Python version info
+            psql -d plpython_testdb -c "
+            CREATE OR REPLACE FUNCTION py_version() RETURNS text AS \$\$
+            import sys
+            return f'Python {sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}'
+            \$\$ LANGUAGE plpython3u;" 2>&1
+
+            # Get the Python version from plpython
+            PLPYTHON_VERSION=$(psql -t -A -d plpython_testdb -c "SELECT py_version();")
+
+            # Get the system Python version
+            SYSTEM_VERSION=$(python --version)
```

Comment:
I was given the feedback that we should always use `sys.version_info` for comparisons, because we can't guarantee that the `--version` output will be stable. You could just run the same code you're doing above here, like in this example:
https://github.com/wolfi-dev/os/blob/f79bd298614f506707b99320ab727f2d0dd24a85/mlflow.yaml#L198

Of course, you could also just compare [ "$PLPYTHON_VERSION" = "${{vars.py-version}}" ] here and avoid the whole system version detection. We're relying on that macro already to install the system python, so installing it then querying it kind of "the long way around".